internal/controllers/nmc_reconciler_test.go

```diff
@@ -217,10 +243,10 @@ var _ = Describe("NodeModulesConfigReconciler_Reconcile", func() {
 			wh.EXPECT().ProcessModuleSpec(contextWithValueMatch, nmc, &spec0, &status0).Return(fmt.Errorf("some error")),
 			wh.EXPECT().ProcessUnconfiguredModuleStatus(contextWithValueMatch, nmc, &status2).Return(fmt.Errorf("some error")),
 			wh.EXPECT().GarbageCollectInUseLabels(ctx, nmc).Return(fmt.Errorf("some error")),
-			wh.EXPECT().UpdateNodeLabelsAndRecordEvents(ctx, nmc).Return(fmt.Errorf("some error")),
+			wh.EXPECT().UpdateNodeLabels(ctx, nmc, &node).Return(nil, nil, fmt.Errorf("some error")),
 		)
 
-		_, err := r.Reconcile(ctx, req)
+		_, err = r.Reconcile(ctx, req)
 		Expect(err).ToNot(BeNil())
```

Comment:
This comment is for purpose learning only as it is not directly related to your PR.

This is not a great validation. The description of the spec is `should complete all the reconcile functions and return combined error`, therefore, the validation should be to (first return different error messages and then) validate that the combined error contain each one of the error messages of the individual errors.

Reply:
You don't have to add it to the PR so we can merge it faster. You can fix it in a future PR if you like. But to answer your question, the answer is YES.

One way to check it is like this example:
https://github.com/kubernetes-sigs/kernel-module-management/blob/c44a5e8d908a3bdd039f1f5cb8ee18663499eb6e/internal/registry/registry_test.go#L57